server/extension/extension.go

```diff
@@ -277,81 +402,106 @@ func (m *Manager) registerExtensions(r *mux.Router, extConfigs *ExtensionConfigs
 	return nil
 }
 
+// authorize will enforce rbac rules are satified for the given RequestResources.
+// The following validations are executed:
+//   - enforce the subject has permission to read application/project provided
+//     in HeaderArgoCDApplicationName and HeaderArgoCDProjectName.
+//   - enforce the subject has permission to invoke the extension identified by
+//     extName.
+//   - enforce that all resources provided in the HeaderArgoCDResourceGVKName belong
+//     to the application identified by HeaderArgoCDApplicationName.
+//
+// If all validations are satified it will return the Application resource
+func (m *Manager) authorize(ctx context.Context, rr *RequestResources, extName string) (*v1alpha1.Application, error) {
+	if m.rbac == nil {
+		return nil, fmt.Errorf("rbac enforcer not set in extension manager")
+	}
+	appRBACName := security.AppRBACName(rr.ApplicationNamespace, rr.ProjectName, rr.ApplicationNamespace, rr.ApplicationName)
+	if err := m.rbac.EnforceErr(ctx.Value(session.CtxKeyClaims), rbacpolicy.ResourceApplications, rbacpolicy.ActionGet, appRBACName); err != nil {
+		return nil, fmt.Errorf("application authorization error: %s", err)
+	}
+
+	if err := m.rbac.EnforceErr(ctx.Value(session.CtxKeyClaims), rbacpolicy.ResourceExtensions, rbacpolicy.ActionInvoke, extName); err != nil {
+		return nil, fmt.Errorf("unauthorized to invoke extension %q: %s", extName, err)
+	}
+
+	// just retrieve the app after checking if subject has access to it
+	app, err := m.application.Get(rr.ApplicationNamespace, rr.ApplicationName)
+	if err != nil {
+		return nil, fmt.Errorf("error getting application: %s", err)
+	}
+	if app == nil {
+		return nil, fmt.Errorf("invalid Application provided in the %q header", HeaderArgoCDApplicationName)
+	}
+
+	// if resources are provided in the HeaderArgoCDResourceGVKName
+	// it will validate if it belongs to the application
+	for _, resource := range rr.Resources {
+		found := false
+		for _, child := range app.Status.Resources {
```

Comment:
I don't think looking at `app.Status.Resources` is enough. If I recall correctly, that list contains only resources which are represented in git (like Deployments). It won't include resources which are included via owner references. 

See `getAppLiveResource` in server/application/application.go for an example of a broader search.

Reply:
Yes, I am aware but that will require additional calls to be made. I intentionally left it this way as a simpler and much lighter implementation. In your example, users will still be able to validate the deployment ownership.

Reply:
Agreed. Removing the `Argocd-Resource-GVK-Name`

Reply:
done